File: app/src/main/java/com/ruideraj/secretelephant/injection/SendModule.java
package com.ruideraj.secretelephant.injection;

import android.content.Context;
import android.telephony.SmsManager;

import com.ruideraj.secretelephant.PropertiesReader;
import com.ruideraj.secretelephant.Runner;
import com.ruideraj.secretelephant.send.EmailSender;
import com.ruideraj.secretelephant.send.SendRepository;
import com.ruideraj.secretelephant.send.SmsSender;

import dagger.Module;
import dagger.Provides;

@Module
public class SendModule {

    @Provides
    static SmsSender providesSmsSender() {
        return new SmsSender(SmsManager.getDefault());
    }

    @Provides
    static EmailSender providesEmailSender(Context context) {
        return new EmailSender(context);
    }

    @Provides
    static PropertiesReader providesPropertiesReader(Context context) {
        return new PropertiesReader(context);
    }

}

File: app/src/main/java/com/ruideraj/secretelephant/injection/SendModule.kt
package com.ruideraj.secretelephant.injection

import android.telephony.SmsManager
import com.ruideraj.secretelephant.PropertiesReader
import com.ruideraj.secretelephant.PropertiesReaderImpl
import com.ruideraj.secretelephant.send.*
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class SendModule {

    @Provides
    fun providesSendRepository(sendRepositoryImpl: SendRepositoryImpl): SendRepository = sendRepositoryImpl

    @Provides
    @Named("SmsSender")
    fun providesSmsSender(propertiesReader: PropertiesReader): Sender = SmsSender(SmsManager.getDefault(),
                                                                                  propertiesReader)

    @Provides
    fun providesEmailSender(emailSenderImpl: EmailSenderImpl): EmailSender = emailSenderImpl

    @Provides
    fun providesPropertiesReader(propertiesReaderImpl: PropertiesReaderImpl): PropertiesReader = propertiesReaderImpl

}